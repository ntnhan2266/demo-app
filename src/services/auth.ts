import { IUser } from '@/interfaces/user';
import { saveAuthUserToLocalStorage } from '@/utils/auth-storage';
import {
  loginUser as mockLoginUser,
  registerUser as mockRegisterUser,
  updateUser as mockUpdateUser,
  updatePassword as mockUpdatePassword,
} from '@/utils/mock-user-database';

export const registerUser = (user: IUser): boolean => {
  // Just call mock function to simulate api call
  return mockRegisterUser(user);
};

export const loginUser = (emailOrPhone: string, password: string): IUser | null => {
  const matchedUser = mockLoginUser(emailOrPhone, password);

  if (matchedUser) {
    // Save auth user info
    saveAuthUserToLocalStorage(matchedUser);

    // Return user info upon successful login
    return matchedUser;
  }

  return null; // Return null for unsuccessful login
};

export const updateUserInfo = (emailOrPhone: string, updatedInfo: Partial<IUser>): IUser | null => {
  // Just call mock function to simulate api call
  const updatedUser = mockUpdateUser(emailOrPhone, updatedInfo);

  if (updatedUser) {
    // Update auth user info
    saveAuthUserToLocalStorage(updatedUser);

    // Return user info upon successful login
    return updatedUser;
  }

  return null;
};

export const updateUserPassword = (emailOrPhone: string, oldPassword: string, newPassword: string): boolean => {
  // Just call mock function to simulate api call
  return mockUpdatePassword(emailOrPhone, oldPassword, newPassword);
};
