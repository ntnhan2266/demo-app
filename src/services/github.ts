import { IGetGithubReposParams, IGithubRepo } from '@/interfaces/api/github';
import createAxiosInstance from '@/services/http-client';
import { snakeCaseKeys } from '@/utils/convert';

const API_PATH = 'https://api.github.com';

const apiClient = createAxiosInstance({
  baseURL: API_PATH,
  headers: {
    Accept: 'application/vnd.github+json',
  },
});

export const getListReposByUsername = async (params: IGetGithubReposParams) => {
  try {
    // Destructure the params
    const { username, page = 1, perPage = 20, sort = 'updated_at', direction = 'desc' } = params;

    // Make the request to the GitHub API
    const response = await apiClient.get(`/users/${username}/repos`, {
      // Adjusting the parameter name based on the interface
      params: snakeCaseKeys({
        page,
        perPage,
        sort,
        direction,
      }),
    });

    // Convert to Array<IGithubRepo>
    const repos: Array<IGithubRepo> = response.data as Array<IGithubRepo>;
    return repos;
  } catch (error) {
    // Handle errors here, e.g., log the error, show a user-friendly message, etc.
    console.error('Error fetching GitHub repositories:', error);
    // Rethrow the error for the calling code to handle
    throw error;
  }
};
