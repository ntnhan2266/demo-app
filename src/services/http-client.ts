import { camelCaseKeys } from '@/utils/convert';
import axios, { AxiosInstance } from 'axios';

// Timeout 1 minute
const REQUEST_TIMEOUT = 60 * 1000;

interface IAxiosInstanceOptions {
  baseURL: string;
  headers?: Record<string, string>;
}

/**
 * Create and configure an Axios instance with interceptors for handling requests and responses.
 * @param {AxiosInstanceOptions} options - Configuration options for the Axios instance.
 * @returns {AxiosInstance} - Configured Axios instance.
 */
const createAxiosInstance = ({ baseURL = '', headers }: IAxiosInstanceOptions): AxiosInstance => {
  const instance = axios.create({
    baseURL,
    timeout: REQUEST_TIMEOUT,
    headers: {
      'Content-Type': 'application/json',
      ...headers,
    },
  });

  instance.interceptors.request.use(
    (config) => {
      // Modify the request config here, such as adding authentication headers
      // config.headers.Authorization = `Bearer ${getToken()}`;
      return config;
    },
    (error) => {
      return Promise.reject(error);
    },
  );

  instance.interceptors.response.use(
    (response) => {
      // Modify the response here before it reaches the calling component
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const modifiedResponseData = camelCaseKeys(response.data as Record<string, any>);
      return { ...response, data: modifiedResponseData };
    },
    (error) => {
      // You can handle errors globally here, such as redirecting to an error page
      return Promise.reject(error);
    },
  );

  return instance;
};

export default createAxiosInstance;
