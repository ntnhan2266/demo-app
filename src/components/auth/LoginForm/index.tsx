import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';
import { loginUser } from '@/services/auth';
import Input from '@/components/common/Input';
import Button from '@/components/common/Button';
import { ROUTE_PATH } from '@/constants/route-path';
import { isValidContact } from '@/utils/validation';
import { useUser } from '@/hooks/useUser';

interface ILoginFormInputs {
  emailOrPhone: string;
  password: string;
}

const LoginForm: React.FC = (): React.ReactElement => {
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<ILoginFormInputs>();
  const { setUser } = useUser();
  const [successMessage, setSuccessMessage] = useState<string | null>(null);
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  const onSubmit = (values: ILoginFormInputs): void => {
    const user = loginUser(values.emailOrPhone, values.password);

    if (!user) {
      setErrorMessage('Login failed. Invalid credentials.');
      return;
    }

    // For simplicity, just display a success message
    setSuccessMessage('Login successful!');

    // Update authenticated user data into context
    setUser(user);

    // Reset the form
    // Note: Use reset() if you need to clear the form fields
    reset();

    // Navigate to the dashboard or any other page after login
    navigate(ROUTE_PATH.DASHBOARD_UPDATE_INFO);
  };

  return (
    <div data-testid='login-form' className='max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow'>
      <h2 className='text-2xl font-bold mb-4 uppercase'>Login to your account</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('emailOrPhone', {
            required: 'Email or phone is required.',
            validate: (val: string) => isValidContact(val) || 'Invalid email or phone.',
          })}
          errorMessage={errors.emailOrPhone?.message}
          type='text'
          label='Email or phone'
          placeholder='Enter your email or phone number'
          wrapperClass='mb-4'
          data-testid='email-or-phone-input'
          inputClasses='w-full md:w-full'
          id='email'
          required
        />
        <Input
          {...register('password', {
            required: 'Password is required.',
          })}
          errorMessage={errors.password?.message}
          type='password'
          label='Password'
          placeholder='Enter your password'
          wrapperClass='mb-4'
          data-testid='password-input'
          inputClasses='w-full md:w-full'
          id='password'
          required
        />
        <Button data-testid='submit-button' label='Login' type='submit' fullWidth />
      </form>
      {successMessage && (
        <div data-testid className='mt-4 p-3 bg-green-200 text-green-800 rounded-md'>
          {successMessage}
        </div>
      )}
      {errorMessage && (
        <div data-testid='error-message' className='mt-4 p-3 bg-red-200 text-red-800 rounded-md'>
          {errorMessage}
        </div>
      )}
    </div>
  );
};

export default LoginForm;
