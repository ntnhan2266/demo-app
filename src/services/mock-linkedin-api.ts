// This file provides a mock implementation of the LinkedIn API service.
// It simulates the behavior of making API calls to LinkedIn.
// Service level

import { IGithubRepo } from '@/interfaces/api/github';
import { IMockResponse } from '@/interfaces/api/linkedin';

export const shareLinkToLinkedin = (repo: IGithubRepo): Promise<IMockResponse> => {
  const successRate = 0.95;

  return new Promise((resolve, reject) => {
    // Simulate a successful response in 95% of cases
    if (Math.random() < successRate) {
      const IMockResponse: IMockResponse = {
        success: true,
        data: {
          repoId: repo.id,
          message: 'Mock LinkedIn API success',
        },
      };
      resolve(IMockResponse);
    } else {
      const mockError: IMockResponse = {
        success: false,
        error: {
          message: 'Mock LinkedIn API error',
        },
      };
      reject(mockError);
    }
  });
};
