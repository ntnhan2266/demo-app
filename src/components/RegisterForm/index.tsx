import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useNavigate } from "react-router-dom";
import Button from '@/components/Button';
import Input from '@/components/Input';
import { isValidContact } from '@/utils/validation';
import { registerUser } from '@/services/auth';
import { ROUTE_PATH } from '@/constants/route-path';

interface IFormInputs {
  firstName: string;
  lastName: string;
  emailOrPhone: string;
  password: string;
  confirmPassword: string;
}

const RegisterForm: React.FC = (): React.ReactElement => {
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    reset,
    watch,
    setError,
    formState: { errors },
  } = useForm<IFormInputs>({
    defaultValues: {
      firstName: '',
      lastName: '',
      emailOrPhone: '',
      password: '',
      confirmPassword: '',
    },
  });
  const [outcomeMessage, setOutcomeMessage] = useState<string | null>(null);

  const onSubmit = (values: IFormInputs): void => {
    const result = registerUser(values);
    if (!result) {
      setError('emailOrPhone', {
        message: 'This email or phone is already in use.'
      });
      return;
    }

    // For simplicity, just display a success message
    setOutcomeMessage('Registration successful!');

    // Reset the form
    reset();

    // Navigate to login page
    navigate(ROUTE_PATH.LOGIN);
  };

  return (
    <div className='max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow-md'>
      <h2 className='text-2xl font-bold mb-4'>Create your account</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('firstName', { required: 'First name is required.' })}
          errorMessage={errors.firstName?.message}
          type='text'
          label='First name'
          placeholder='Enter your first name'
          required
        />
        <Input
          {...register('lastName', { required: 'Last name is required.' })}
          errorMessage={errors.lastName?.message}
          type='text'
          label='Last name'
          placeholder='Enter your last name'
          required
        />
        <Input
          {...register('emailOrPhone', {
            required: 'Email or phone is required.',
            validate: (val: string) => isValidContact(val) || 'Invalid email or phone.',
          })}
          errorMessage={errors.emailOrPhone?.message}
          type='text'
          name='emailOrPhone'
          label='Email or phone'
          placeholder='Enter either your email or phone number'
          required
        />
        <Input
          {...register('password', {
            required: 'Password is required',
            pattern: {
              value: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{12,}$/,
              message:
                'Password must be strong (include uppercase, lowercase, number, special character, and at least 12 characters).',
            },
          })}
          errorMessage={errors.password?.message}
          type='password'
          label='Password'
          placeholder='Enter your password'
          required
        />
        <Input
          {...register('confirmPassword', {
            required: 'Confirm password is required.',
            validate: (val: string) => {
              if (watch('password') != val) {
                return 'Your passwords do not match.';
              }
            },
          })}
          errorMessage={errors.confirmPassword?.message}
          type='password'
          label='Confirm password'
          placeholder='Confirm your password'
          required
        />
        <Button label='Register' type='submit' fullWidth />
      </form>
      {outcomeMessage && <div className='mt-4 p-3 bg-green-200 text-green-800 rounded-md'>{outcomeMessage}</div>}
    </div>
  );
};

export default RegisterForm;
