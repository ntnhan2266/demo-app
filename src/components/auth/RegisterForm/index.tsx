import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import Button from '@/components/common/Button';
import Input from '@/components/common/Input';
import { isValidContact } from '@/utils/validation';
import { registerUser } from '@/services/auth';
import ErrorModal from '@/components/common/ErrorModal';
import SuccessModal from '@/components/common/SuccessModal';

interface IFormInputs {
  firstName: string;
  lastName: string;
  emailOrPhone: string;
  password: string;
  confirmPassword: string;
}

const RegisterForm: React.FC = (): React.ReactElement => {
  const {
    register,
    handleSubmit,
    reset,
    watch,
    setError,
    formState: { errors },
  } = useForm<IFormInputs>({
    defaultValues: {
      firstName: '',
      lastName: '',
      emailOrPhone: '',
      password: '',
      confirmPassword: '',
    },
  });
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
  const [isSuccessModalOpen, setIsSuccessModalOpen] = useState(false);

  const closeErrorModal = (): void => {
    // Close the error modal
    setIsErrorModalOpen(false);
  };

  const closeSuccessModal = (): void => {
    // Close the success modal
    setIsSuccessModalOpen(false);
  };

  const onSubmit = (values: IFormInputs): void => {
    const result = registerUser(values);
    if (!result) {
      // Registration failed
      setIsErrorModalOpen(true);
      setError('emailOrPhone', {
        message: 'This email or phone is already in use.',
      });
      return;
    }

    // Registration successful
    setIsSuccessModalOpen(true);

    // Reset the form
    reset();
  };

  return (
    <div data-testid='register-form' className='max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow'>
      <h2 className='text-2xl font-bold mb-4 uppercase'>Create your account</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('firstName', { required: 'First name is required.' })}
          errorMessage={errors.firstName?.message}
          type='text'
          label='First name'
          placeholder='Enter your first name'
          wrapperClass='mb-4'
          data-testid='first-name-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('lastName', { required: 'Last name is required.' })}
          errorMessage={errors.lastName?.message}
          type='text'
          label='Last name'
          placeholder='Enter your last name'
          wrapperClass='mb-4'
          data-testid='last-name-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('emailOrPhone', {
            required: 'Email or phone is required.',
            validate: (val: string) => isValidContact(val) || 'Invalid email or phone.',
          })}
          errorMessage={errors.emailOrPhone?.message}
          type='text'
          name='emailOrPhone'
          label='Email or phone'
          placeholder='Enter either your email or phone number'
          wrapperClass='mb-4'
          data-testid='email-or-phone-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('password', {
            required: 'Password is required.',
            pattern: {
              value: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*(),.?":{}|<>])[A-Za-z\d!@#$%^&*(),.?":{}|<>]{12,}$/,
              message:
                'Password must be strong (include uppercase, lowercase, number, special character, and at least 12 characters).',
            },
          })}
          errorMessage={errors.password?.message}
          type='password'
          label='Password'
          placeholder='Enter your password'
          wrapperClass='mb-4'
          data-testid='password-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('confirmPassword', {
            required: 'Confirm password is required.',
            validate: (val: string) => {
              if (watch('password') != val) {
                return 'Your passwords do not match.';
              }
            },
          })}
          errorMessage={errors.confirmPassword?.message}
          type='password'
          label='Confirm password'
          placeholder='Confirm your password'
          wrapperClass='mb-4'
          data-testid='confirm-password-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Button data-testid='submit-button' label='Register' type='submit' fullWidth />
      </form>
      <ErrorModal
        title='Failure!'
        content='There was an error during the registration process. Please try again.'
        isOpen={isErrorModalOpen}
        onClose={closeErrorModal}
      />
      <SuccessModal
        title='Success!'
        content='Your account has been successfully registered.'
        isOpen={isSuccessModalOpen}
        onClose={closeSuccessModal}
      />
    </div>
  );
};

export default RegisterForm;
