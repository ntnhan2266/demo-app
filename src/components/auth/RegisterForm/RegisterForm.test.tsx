import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import RegisterForm from '@/components/auth/RegisterForm';
import { registerUser } from '@/services/auth';
import renderer from 'react-test-renderer';

// Mock the services and functions
jest.mock('@/services/auth', () => ({
  registerUser: jest.fn(),
}));

describe('RegisterForm', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    render(<RegisterForm />);
  });

  test('should match the snapshot', () => {
    const tree = renderer.create(<RegisterForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render the form with the correct inputs', () => {
    const firstNameInput = screen.getByTestId('first-name-input');
    const lastNameInput = screen.getByTestId('last-name-input');
    const emailOrPhoneInput = screen.getByTestId('email-or-phone-input');
    const passwordInput = screen.getByTestId('password-input');
    const confirmPasswordInput = screen.getByTestId('confirm-password-input');
    const submitButton = screen.getByTestId('submit-button');

    expect(firstNameInput).toBeInTheDocument();
    expect(lastNameInput).toBeInTheDocument();
    expect(emailOrPhoneInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(confirmPasswordInput).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
  });

  test('should display error messages when form is submitted with invalid data', async () => {
    const submitButton = screen.getByTestId('submit-button');
    fireEvent.click(submitButton);

    // Ensure that error messages are displayed for required fields
    expect(await screen.findByText('First name is required.')).toBeInTheDocument();
    expect(await screen.findByText('Last name is required.')).toBeInTheDocument();
    expect(await screen.findByText('Email or phone is required.')).toBeInTheDocument();
    expect(await screen.findByText('Password is required.')).toBeInTheDocument();
    expect(await screen.findByText('Confirm password is required.')).toBeInTheDocument();
  });

  test('should display error message when email or phone is invalid', async () => {
    const emailOrPhoneInput = screen.getByTestId('email-or-phone-input');
    fireEvent.input(emailOrPhoneInput, { target: { value: 'invalid-email' } });

    const submitButton = screen.getByTestId('submit-button');
    fireEvent.click(submitButton);

    // Ensure that an error message for invalid email/phone is displayed
    expect(await screen.findByText('Invalid email or phone.')).toBeInTheDocument();
  });

  test('should display error message when passwords do not match', async () => {
    const passwordInput = screen.getByTestId('password-input');
    const confirmPasswordInput = screen.getByTestId('confirm-password-input');

    fireEvent.input(passwordInput, { target: { value: 'qwerQWER1234!@' } });
    fireEvent.input(confirmPasswordInput, { target: { value: 'qwerQWER1234!@@' } });

    const submitButton = screen.getByTestId('submit-button');
    fireEvent.click(submitButton);

    // Ensure that an error message for passwords not matching is displayed
    expect(await screen.findByText('Your passwords do not match.')).toBeInTheDocument();
  });

  test('should call registerUser function when form is submitted with valid data', async () => {
    (registerUser as jest.Mock).mockReturnValue(true);

    const firstNameInput = screen.getByTestId('first-name-input');
    const lastNameInput = screen.getByTestId('last-name-input');
    const emailOrPhoneInput = screen.getByTestId('email-or-phone-input');
    const passwordInput = screen.getByTestId('password-input');
    const confirmPasswordInput = screen.getByTestId('confirm-password-input');
    const submitButton = screen.getByTestId('submit-button');

    fireEvent.input(firstNameInput, { target: { value: 'John' } });
    fireEvent.input(lastNameInput, { target: { value: 'Doe' } });
    fireEvent.input(emailOrPhoneInput, { target: { value: 'test@example.com' } });
    fireEvent.input(passwordInput, { target: { value: 'StrongPassword123!@' } });
    fireEvent.input(confirmPasswordInput, { target: { value: 'StrongPassword123!@' } });

    fireEvent.click(submitButton);

    // Wait for the success modal to appear
    await waitFor(() => {
      expect(registerUser).toHaveBeenCalledWith({
        firstName: 'John',
        lastName: 'Doe',
        emailOrPhone: 'test@example.com',
        password: 'StrongPassword123!@',
        confirmPassword: 'StrongPassword123!@',
      });
      expect(screen.getByText('Your account has been successfully registered.')).toBeInTheDocument();
    });
  });

  test('should display error modal when registration fails', async () => {
    (registerUser as jest.Mock).mockReturnValue(false);

    const firstNameInput = screen.getByTestId('first-name-input');
    const lastNameInput = screen.getByTestId('last-name-input');
    const emailOrPhoneInput = screen.getByTestId('email-or-phone-input');
    const passwordInput = screen.getByTestId('password-input');
    const confirmPasswordInput = screen.getByTestId('confirm-password-input');
    const submitButton = screen.getByTestId('submit-button');

    fireEvent.input(firstNameInput, { target: { value: 'John' } });
    fireEvent.input(lastNameInput, { target: { value: 'Doe' } });
    fireEvent.input(emailOrPhoneInput, { target: { value: 'test@example.com' } });
    fireEvent.input(passwordInput, { target: { value: 'StrongPassword123!@' } });
    fireEvent.input(confirmPasswordInput, { target: { value: 'StrongPassword123!@' } });

    fireEvent.click(submitButton);

    // Wait for the error modal to appear
    await waitFor(() => {
      expect(screen.getByText('This email or phone is already in use.')).toBeInTheDocument();
    });
  });
});
