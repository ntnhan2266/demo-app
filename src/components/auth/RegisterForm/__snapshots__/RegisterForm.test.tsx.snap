// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should match the snapshot 1`] = `
<div
  className="max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow-md"
  data-testid="register-form"
>
  <h2
    className="text-2xl font-bold mb-4"
  >
    Create your account
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="First name"
      >
        First name
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="first-name-input"
        disabled={false}
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your first name"
        type="text"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Last name"
      >
        Last name
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="last-name-input"
        disabled={false}
        name="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your last name"
        type="text"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Email or phone"
      >
        Email or phone
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="email-or-phone-input"
        disabled={false}
        name="emailOrPhone"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter either your email or phone number"
        type="text"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Password"
      >
        Password
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="password-input"
        disabled={false}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Confirm password"
      >
        Confirm password
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="confirm-password-input"
        disabled={false}
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm your password"
        type="password"
      />
    </div>
    <button
      className="px-4 py-2 rounded-md focus:outline-none  bg-blue-500 text-white w-full"
      data-testid="submit-button"
      disabled={false}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
